YARP formula failing on Travis

In several macOS-based Travis builds we are experiencing the following failure of the yarp formula: 
~~~
==> Installing robotology/formulae/yarp
==> Downloading https://github.com/robotology/yarp/archive/v2.3.70.1.tar.gz
==> Downloading from https://codeload.github.com/robotology/yarp/tar.gz/v2.3.70.
==> cmake -DCMAKE_C_FLAGS_RELEASE=-DNDEBUG -DCMAKE_CXX_FLAGS_RELEASE=-DNDEBUG -D
Last 15 lines from /Users/travis/Library/Logs/Homebrew/yarp/01.cmake:
  Tried extensions .c .C .c++ .cc .cpp .cxx .m .M .mm .h .hh .h++ .hm .hpp
  .hxx .in .txx
CMake Error in src/yarpmotorgui/CMakeLists.txt:
  Cannot find source file:
main.cpp;flowlayout.cpp;jointitem.cpp;mainwindow.cpp;partitem.cpp;piddlg.cpp;sequencewindow.cpp;sliderOptions.cpp;sliderWithTarget.cpp;startdlg.cpp;flowlayout.h;jointitem.h;log.h;mainwindow.h;partitem.h;piddlg.h;sliderOptions.h;sliderWithTarget.h;yarpmotorgui.h;sequencewindow.h;startdlg.h;res.qrc;/tmp/yarp-20171122-14172-1efr2r7/yarp-2.3.70.1/src/yarpmotorgui/qrc_res.cpp;jointitem.ui;mainwindow.ui;piddlg.ui;sequencewindow.ui;startdlg.ui;sliderOptions.ui;/tmp/yarp-20171122-14172-1efr2r7/yarp-2.3.70.1/src/yarpmotorgui/ui_jointitem.h;/tmp/yarp-20171122-14172-1efr2r7/yarp-2.3.70.1/src/yarpmotorgui/ui_mainwindow.h;/tmp/yarp-20171122-14172-1efr2r7/yarp-2.3.70.1/src/yarpmotorgui/ui_piddlg.h;/tmp/yarp-20171122-14172-1efr2r7/yarp-2.3.70.1/src/yarpmotorgui/ui_sequencewindow.h;/tmp/yarp-20171122-14172-1efr2r7/yarp-2.3.70.1/src/yarpmotorgui/ui_startdlg.h;/tmp/yarp-20171122-14172-1efr2r7/yarp-2.3.70.1/src/yarpmotorgui/ui_sliderOptions.h

  Tried extensions .c .C .c++ .cc .cpp .cxx .m .M .mm .h .hh .h++ .hm .hpp
  .hxx .in .txx

-- Generating done
-- Build files have been written to: /tmp/yarp-20171122-14172-1efr2r7/yarp-2.3.70.1
If reporting this issue please do so at (not Homebrew/brew or Homebrew/core):
https://github.com/robotology/homebrew-formulae/issues
~~~

With `brew install robotology/formulae/yarp --HEAD` I get: 

```
CMake Error at cmake/YarpFindDependencies.cmake:162 (message):
  Optional package RTF not found.  Please install it or disable the option
  "YARP_COMPILE_RTF_ADDONS" to build yarp.
Call Stack (most recent call first):
  cmake/YarpFindDependencies.cmake:532 (check_optional_dependency)
  CMakeLists.txt:71 (include)
```

Don't we have two different problems here?

Problem 1:
> I'm having the same problem when issuing brew install robotology/formulae/yarp on macOS High Sierra 10.13.1. and CMake 3.10.0. What was the way to proceed at this point?

Was fixed with this PR https://github.com/robotology/yarp/pull/1465 and with a new YARP release we will be able to fix the formula.

Problem 2:



> With brew install robotology/formulae/yarp --HEAD I get:
> 
> CMake Error at cmake/YarpFindDependencies.cmake:162 (message):
>   Optional package RTF not found.  Please install it or disable the option
>   "YARP_COMPILE_RTF_ADDONS" to build yarp.
> Call Stack (most recent call first):
>   cmake/YarpFindDependencies.cmake:532 (check_optional_dependency)
>   CMakeLists.txt:71 (include)

This may be a configuration option that needs further investigations.

--- a/Formula/yarp.rb
+++ b/Formula/yarp.rb
@@ -8,8 +8,8 @@ class Yarp < Formula
   end
 
   bottle do
-    root_url "https://github.com/robotology/yarp/releases/download/v2.3.70.1"
-    sha256 "f932056d3f1b2cac2cad6c8695f807e142bea0f1927e4507ec616ef5524ccbc9" => :sierra
+    root_url "https://github.com/robotology/yarp/releases/download/v2.3.70.2"
+    sha256 "d68bd7e09a45f3b066f1556c8016d3ad3b1f20ebd1c2b4868a75171aa14c0564" => :sierra
   end
 
   head do

--- a/Formula/yarp.rb
+++ b/Formula/yarp.rb
@@ -3,8 +3,8 @@ class Yarp < Formula
   homepage "http://yarp.it"
 
   stable do
-    url "https://github.com/robotology/yarp/archive/v2.3.70.1.tar.gz"
-    sha256 "c693956aa76082bff18645a4a0cadf43d564c82fe518420e2b6c8410f716bf85"
+    url "https://github.com/robotology/yarp/archive/v2.3.70.2.tar.gz"
+    sha256 "888c4d2d85e0d9a3da4c4a111ed96c77c35a2849ed9bf079d69b08941f40ea79"
   end
 
   bottle do

--- a/Formula/yarp.rb
+++ b/Formula/yarp.rb
@@ -13,7 +13,7 @@ class Yarp < Formula
   end
 
   head do
-      url "https://github.com/robotology/yarp.git", :branch => "devel"
+      url "https://github.com/robotology/yarp.git", :branch => "master"
   end
 
   option "without-qt5", "Build without GUI applications"

--- a/Formula/yarp.rb
+++ b/Formula/yarp.rb
@@ -50,7 +50,7 @@ class Yarp < Formula
 
     args << "-DCREATE_GUIS:BOOL=TRUE" if build.with? "qt"
     if build.with? "opencv"
-      args << "-DENABLE_yarpmod_opencv_grabber:BOOL=TRUE" 
+      args << "-DENABLE_yarpmod_opencv_grabber:BOOL=TRUE"
     else
       args << "-DCMAKE_DISABLE_FIND_PACKAGE_OpenCV:BOOL=TRUE"
     end

